core/src/main/java/org/mule/api/metadata/MuleMetadataManager.java

```diff
@@ -66,6 +72,30 @@
                                              String.format(EXCEPTION_RESOLVING_OPERATION_METADATA, componentId));
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void disposeCache(String id)
+    {
+        caches.remove(id);
+    }
+
+    public MetadataCache getCache(String id)
+    {
+        MetadataCache cache = caches.get(id);
```

Comment:
this is not thread safe and makes no sense to code that.. use a guava LoadingCache instead